homeworks/06-advanced-redux/src/hooks/useApp.ts

```diff
@@ -1,29 +1,45 @@
 import React from "react";
-import {Pizza} from "../types";
-import {getPizza} from "../services/api";
+import {Pizza, PizzaId, PizzaPrice, State} from "../types";
+import {getPizzas} from "../services/api";
 import * as R from "ramda";
+import {thunkLoadPizzas} from "../store/pizza-list-reducer";
+import {store} from "../store";
+import {ThunkDispatch} from "redux-thunk";
+import {AnyAction} from "redux";
+import {addPizzaToBasket, removePizzaFromBasket, viewPizzas} from "../store/actions";
+
+const dispatch = store.dispatch as ThunkDispatch<State, {}, AnyAction>
 
 export function useApp() {
```

Comment:
The task doesn't mention the use of hooks to create local state. State is provided by Redux Store as passed to App via provider.
Then you use [mapStateToProps](https://react-redux.js.org/using-react-redux/connect-mapstate) to connect to store and [mapDispatchToProps](https://react-redux.js.org/using-react-redux/connect-mapdispatch) to execute dispatch actions to store when user clicks the buttons.